### PR TITLE
Add more SBOMs

### PR DESCRIPTION
**Pull request by rzhao271**
I looked through the published pipeline artifacts and noticed that a few SBOMs seemed to be missing.

This PR adds in the SBOMs as part of an S360 work item.

TO-DO:
- [x] Run custom build to ensure that the newer SBOMs are generated properly and contain the correct documentation <- edit: confirmation build https://dev.azure.com/monacotools/Monaco/_build/results?buildId=237746&view=results

### CODE REVIEW

File: build/azure-pipelines/cli/cli-compile-and-publish.yml [hunk 1]
```diff
@@ -106,3 +106,25 @@ steps:
...
+        displayName: Make folder for SBOM (Windows)
+
+    - ${{ else }}:
+      - script: mkdir $(Build.ArtifactStagingDirectory)/sbom_${{ parameters.VSCODE_CLI_ARTIFACT }}
+        displayName: Make folder for SBOM (non-Windows)
+
+    # The if cases above are for different OSes,
+    # but we're still in the branch where the cli is being published in general.
+    # Generate and publish an SBOM.
+    - task: AzureArtifacts.manifest-generator-task.manifest-generator-task.ManifestGeneratorTask@0
```
Comment (connor4312) [inline build/azure-pipelines/cli/cli-compile-and-publish.yml:122]: Does this make an SBOM for the entire client? Should this stage be scoped to only make an SBOM for the ./cli folder?

Comment (rzhao271) [reply build/azure-pipelines/cli/cli-compile-and-publish.yml:122]: I noticed that the win32 client and server SBOMs both span the entire workspace. I can restrict the SBOM to only be generated for the CLI folder, but I wonder whether the server SBOM should also be restricted in scope?

Comment (connor4312) [reply build/azure-pipelines/cli/cli-compile-and-publish.yml:122]: That would make sense to me. I don't think it necessarily _hurts_, it just means any governance alert will ping additional targets as false-positives, but being more tailored would be nice

Comment (rzhao271) [reply build/azure-pipelines/cli/cli-compile-and-publish.yml:122]: Just checked the generated SBOM and you're right that the SBOM should be scoped to only the cli folder. I'll also see if I can restrict the server SBOM.

**Review by connor4312 — APPROVED** [review]

**Comment by smoothzebra** [comment]
ในวันที่ พ. 18 ต.ค. 2023 05:56 น. Raymond Zhao ***@***.***>
เขียนว่า:

**Pull request merged**